telemetry/specs/telemetry_report.mdk

```diff
@@ -496,41 +577,103 @@ particular deployment.
-    Specific Extension Data, and excluding the 4-byte word (RepType, InType, Report Length, 
-    MD Length, D, Q, F, I, Rsvd).
+  : Indicates the length of the Individual Report Header in a multiple of
+    4-byte words, including the *Individual Report Main Contents* and
+    *Individual Report Inner Contents*, but excluding the length of the first
+    4-byte word (*RepType, InType, Report Length, MD Length, D, Q, F, I, Rsvd*).
+
+    For *RepType* codepoint 1 *INT*, this includes the length of *RepMdBits,
+    Domain Specific ID, DSMdBits, DSMdstatus, Variable Optional Baseline
+    Metadata,* and *Variable Optional Domain Specific Metadata* (see Section
+    [#int-report]).
 
 * MD Length (8b): Metadata Length
-  : Indicates the length of Variable Optional Baseline Metadata and Variable Optional 
-    Domain Specific Metadata in 4-byte words.
+  : Indicates the length of metadata included in this report in a multiple
+    of 4-byte words. This may help the telemetry monitoring system
+    determine where the *Individual Report Inner Contents* begins.
+
+    For *RepType* codepoint 1 *INT*, this includes the length of the *Variable
+    Optional Baseline Metadata* and *Variable Optional Domain Specific
+    Metadata* in 4-byte words (see Section [#int-report]).
 
 * D (1b): Dropped
-  : Indicates that at least one packet matching a drop watchlist was dropped.
+  : Indicates that at least one packet matching a watchlist was dropped.
 
 * Q (1b): Congested Queue Association
   : Indicates the presence of congestion on a monitored queue.
 
 * F (1b): Tracked Flow Association
   : Indicates that this telemetry report is for a tracked flow, i.e. the
-    packet matched a flow watchlist somewhere (in case of INT-MD, INT-MX or IOAM) or
+    packet matched a watchlist somewhere (in case of INT-MD, INT-MX or IOAM) or
     locally (in case of INT-XD). The report might include INT-MD or IOAM
     metadata in the truncated packet. Other telemetry reports are
-    likely to be received for the same tracked flow, from the same network
-    device and (in case of drop reports, INT-MX, INT-XD or path changes) from
-    other network devices.
-  
+    likely to be received for the same tracked flow, from the same node
+    and (in case of drop reports, INT-MX, INT-XD or path changes) from
+    other nodes.
+
 * I (1b): Intermediate Report
   : Indicates that a transit node sent this intermediate report for INT-MD.
-  
-* Rsvd (4b): Reserved 
+
+* Rsvd (4b): Reserved
   : Should be set to zero upon transmission, and ignored upon reception
 
+* Individual Report Main Contents
+  : The metadata that comprises this report, along with associated fields
+    that assist in processing the metadata. The format varies depending on
+    *RepType*.
+
+    When the *RepType* value is *Inner Only*, then the Individual Report Main
+    Contents is empty. *MD Length* should be set to zero upon transmission,
```

Comment:
When MD Length is zero, even the RepMdBits and Domain Specific ID, DSMdBits, DSMdStatus fields are also not there. However, the MD Length is defined as the length of the Variable Optional Baseline Metadata and DS Metadata. The 2 quad-byte words corresponding to RepMdBits and Domain Specific ID, DSMdBits, DSMdStatus  are not included in the MD Length. We need to capture this somewhere.

Reply:
Your first sentence is not true. Presence of RepMdBits, Domain Specific ID, DsMdBits, DsMdstatus is based on RepType value of INT.

Reply:
I see. Individual Report Type Contents for RepType value of INT defines those fields. If the RepType value is anything else, then the format is different. Makes sense.

Reply:
Added a sentence clarifying that the length does not include the fixed portion of Individual Report Main Contents.